pagermaid/update/v1402.py

```diff
@@ -1,16 +1,20 @@
+from datetime import datetime
 from pathlib import Path
+import shutil
 
 data = Path("data")
 data.mkdir(exist_ok=True)
 
 
-def rename(old: Path):
+def rename(old: Path, need_backup: bool = True):
     if old.exists():
         old_file_name = old.name
         new = data / old_file_name
-        if new.exists():
-            new.unlink()
-        old.rename(new)
+        if need_backup and new.exists():
+            datetime_str = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+            backup = new.with_stem(f"{new.stem}.{datetime_str}")
```

Comment:
python 3.8 没有 with_stem 方法

```suggestion
            backup = data / (new.stem + f".{datetime_str}" + new.suffix)
```